### PR TITLE
[PIR-1365] Report filter not working correctly

### PR DESCRIPTION
**Pull request by Advent51**
- hyperlinks "Excludes" and "Includes" were fixed and now working
- CONDITION_TYPES.IN was added
- filter title and filter dialog now correspond with the last saved filter state
- unsafe console usages were wrapped

### CODE REVIEW

**Comment by Advent51** [comment]
@dcleao @pamval @mchen-len-son @nantunes please review

**Comment by Advent51** [comment]
Please merge along with https://github.com/pentaho/pentaho-platform-plugin-interactive-reporting/pull/635

File: impl/client/src/main/javascript/web/dojo/pentaho/common/FilterDialog.js [hunk 1]
```diff
@@ -635,29 +643,34 @@ define(["dojo/_base/declare", "dijit/_WidgetBase", "dijit/_TemplatedMixin", "dij
...
+      case pentaho.pda.Column.CONDITION_TYPES.IN:
+        switch (filter.combinationType) {
+          case pentaho.pda.Column.OPERATOR_TYPES.AND:
+            friendlyOperator = this.getLocaleString("FilterCombinationTypeIn");
+            break;
+          case pentaho.pda.Column.OPERATOR_TYPES.AND_NOT:
+            friendlyOperator = this.getLocaleString("FilterCombinationTypeNotIn");
+            break;
+          default:
+            console.log("Unknown filter combination type for IN condition type: " + filter.combinationType);
```
Comment (afrjorge) [inline impl/client/src/main/javascript/web/dojo/pentaho/common/FilterDialog.js]: careful, in some environments `console` might be `undefined`

Comment (Advent51) [reply impl/client/src/main/javascript/web/dojo/pentaho/common/FilterDialog.js]: I used what we already use:
https://github.com/pentaho/pentaho-platform-plugin-common-ui/blob/master/impl/client/src/main/javascript/web/dojo/pentaho/common/FilterDialog.js#L283
https://github.com/pentaho/pentaho-platform-plugin-common-ui/blob/master/impl/client/src/main/javascript/web/dojo/pentaho/common/FilterDialog.js#L320
https://github.com/pentaho/pentaho-platform-plugin-common-ui/blob/master/impl/client/src/main/javascript/web/dojo/pentaho/common/FilterDialog.js#L337
https://github.com/pentaho/pentaho-platform-plugin-common-ui/blob/master/impl/client/src/main/javascript/web/dojo/pentaho/common/FilterDialog.js#L357
https://github.com/pentaho/pentaho-platform-plugin-common-ui/blob/master/impl/client/src/main/javascript/web/dojo/pentaho/common/FilterDialog.js#L627
https://github.com/pentaho/pentaho-platform-plugin-common-ui/blob/master/impl/client/src/main/javascript/web/dojo/pentaho/common/FilterDialog.js#L647

@afrjorge should I remove all of these?

Comment (afrjorge) [reply impl/client/src/main/javascript/web/dojo/pentaho/common/FilterDialog.js]: @pentaho/millenniumfalcon ^

Comment (dcleao) [reply impl/client/src/main/javascript/web/dojo/pentaho/common/FilterDialog.js]: Those references are in default branches, that apparently should never execute. Maybe that' why they didn't raise any problems.

Wrapping in `typeof` test would solve the problem:
```js
if(typeof console !== "undefined") {
  console.log("message");
}
```

**Comment by Advent51** [comment]
@afrjorge @dcleao please review

**Pull request merged**